nats/js/client.py

```diff
@@ -136,6 +185,50 @@ async def publish(
             raise nats.js.errors.APIError.from_error(resp['error'])
         return api.PubAck.from_response(resp)
 
+    async def publish_async(
+        self,
+        subject: str,
+        payload: bytes = b'',
+        stream: Optional[str] = None,
+        headers: Optional[Dict] = None,
+    ) -> asyncio.Future[api.PubAck]:
+        """
+        emits a new message to JetStream and returns a future that can be awaited for acknowledgement.
+        """
+
+        if len(self._paf_map) > self._pending_acks_limit:
+            raise nats.js.errors.TooManyStalledMsgsError
```

Comment:
Here instead of raising the error I think we should wait for a default stall time, then when that times out we would raise the error: 

https://github.com/nats-io/nats.go/blob/main/js.go#L951-L958

once it has been stall for that long then the error would be raised. 
The default stall wait in Go is 200ms but for Python it could be longer, thinking maybe a one second.